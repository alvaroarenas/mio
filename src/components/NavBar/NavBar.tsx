import React from 'react';
import styled from 'styled-components';

const Header = styled.header`
  background: #0000bf;
  position: relative;
  height: 75px;

  display: flex;
  align-items: center;
  justify-content: center;
`;

const H1 = styled.h1`
  display: block;
  margin: 0 0 0 0;
  font-family: Paytone One;
  font-style: normal;
  font-weight: normal;
  font-size: 36px;
  line-height: 68px;

  color: white;
`;

type NavBarProps = {
  title: string;
};

const NavBar: React.FC<NavBarProps> = ({ title }) => {
  return (
    <Header>
      <H1>{title}</H1>
    </Header>
  );
};

export default NavBar;
