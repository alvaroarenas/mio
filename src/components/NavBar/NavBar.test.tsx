import React from 'react';
import { render, screen } from '@testing-library/react';
// import NavBar from './NavBar';
import NavBar from './NavBar';
import 'jest-styled-components';

test('renders without crashing', () => {
  const title = 'Mio';
  render(<NavBar title={title} />);
  screen.getByRole('heading', { name: /mio/i });
});

test('renders the giving title', () => {
  const title = 'This is a new longer title';
  render(<NavBar title={title} />);
  screen.getByRole('heading', { name: /This is a new longer title/ });
});

it('matches the snapshot', () => {
  const { container } = render(<NavBar title={'The title'} />);
  expect(container.firstChild).toMatchSnapshot();
});
