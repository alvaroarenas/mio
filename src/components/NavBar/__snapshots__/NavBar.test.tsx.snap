// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
.c0 {
  background: #0000bf;
  position: relative;
  height: 75px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: block;
  margin: 0 0 0 0;
  font-family: Paytone One;
  font-style: normal;
  font-weight: normal;
  font-size: 36px;
  line-height: 68px;
  color: white;
}

<header
  class="c0"
>
  <h1
    class="c1"
  >
    The title
  </h1>
</header>
`;
